### PR TITLE
Fix on target test issues

### PR DESCRIPTION
**Pull request by RonEld**
Backport PR https://github.com/ARMmbed/mbedtls/pull/2230 to mbed-crypto repository and add a commit from #162 to have the on target tests build and run

### CODE REVIEW

**Comment by RonEld** [comment]
I believe CI failure will be fixed with https://github.com/ARMmbed/mbedtls/pull/2858

File: tests/suites/target_test.function [hunk 1]
```diff
@@ -90,18 +90,19 @@ uint8_t receive_byte()
...
-                           greentea_getc(),
-                           greentea_getc(),
-                           greentea_getc(),
-                           greentea_getc(),
-                           greentea_getc(),
-                           '\0'
-                         };
-    assert( unhexify( &value, c ) != 8 );
-    return( (uint32_t)value );
+    uint8_t c_be[8] = { greentea_getc(),
```
Comment (Patater) [inline tests/suites/target_test.function]: Did we need to drop `const` from `c_be` here? Could we put it back?

**Comment by RonEld** [comment]
@Patater @dgreen-arm I have added the 'const' type. Please review

**Comment by RonEld** [comment]
Backported to the Mbed TLS repo in : https://github.com/ARMmbed/mbedtls/pull/2864 and https://github.com/ARMmbed/mbedtls/pull/2866 ( stack reduction changes only, as On Target Tests not available in `mbedtls-2.7`)

**Pull request merged**